engine/src/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxPoiWorkbook.java

```diff
@@ -125,13 +132,19 @@ public KSheet getSheet( String sheetName ) {
 
   @Override
   public void close() {
-    // close all the sheets
-    for ( StaxPoiSheet sheet : openSheetsMap.values() ) {
-      try {
+    try {
+      // close all the sheets
+      for ( StaxPoiSheet sheet : openSheetsMap.values() ) {
         sheet.close();
-      } catch ( Exception e ) {
-        e.printStackTrace();
       }
+      if ( opcpkg != null ) {
+        //We should not save change in xlsx because it is input step.
+        opcpkg.revert();
+      }
+    } catch ( IOException e ) {
```

Comment:
@SergeyTravin Shouldn't we be doing the try/catch inside the for loop so we still try to close all sheets even if one fails to close?